test-bridge/src/main/scala/org/scalajs/testing/bridge/JSRPC.scala

```diff
@@ -25,13 +25,33 @@ import scala.scalajs.concurrent.JSExecutionContext.Implicits.queue
 
 import scala.concurrent.duration._
 
+import scala.scalajs.LinkingInfo
+import scala.scalajs.LinkingInfo.linkTimeIf
 import org.scalajs.testing.common.RPCCore
 
 /** JS RPC Core. Uses `scalajsCom`. */
 private[bridge] final object JSRPC extends RPCCore {
-  Com.init(handleMessage _)
+  linkTimeIf(LinkingInfo.targetPureWasm) {
+    println("init scalajsCom")
+    ScalajsCom.init()
+  } {
+    Com.init(handleMessage _)
+  }
+
+  // TODO: export only if it requires essential exports
+  @JSExportTopLevel("handle")
+  def handle(msg: String): Unit = {
```

Comment:
Instead of using an `@JSExportTopLevel` for this, you should be able to use a dedicated `SymbolRequirement` in `Emitter.symbolRequirements`. This way you can selectively export it.

Also, instead of actually *exporting* it, you could change the `"essentialExterns" "init"` helper accept a `ref.func` to call. That `ref.func` would be a special function that the emitter generates, and which calls this `handle` method.

Reply:
Ah thanks, I was wondering how to export a specific Scala methods in a form of `ref.func` from `Emitter`, but  `symbolRequirements` is a good idea. 

https://github.com/scala-wasm/scala-wasm/pull/59/commits/ba3c5625da7defe7dd8b2f907528ad6f5f32aad7